src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1293,21 +1301,17 @@ public override object VisitField(FieldSymbol symbol, TypeCompilationState argum
 
                             var factory = new SyntheticBoundNodeFactory(methodSymbol, syntax, compilationState, diagsForCurrentMethod);
 
-                            // Iterators handled in IteratorRewriter.cs
-                            if (!methodSymbol.IsIterator)
-                            {
-                                var boundStatementsWithNullCheck = LocalRewriter.TryConstructNullCheckedStatementList(methodSymbol.Parameters, boundStatements, factory);
+                            var nullCheckStatements = LocalRewriter.TryConstructNullCheckedStatementList(methodSymbol.Parameters, factory);
```

Comment:
>TryConstructNullCheckedStatementList [](http://example.com/codeflow?start=68&length=36)

Consider having `TryConstructNullCheckedStatementList()` return `ImmutableArray<BoundStatement>.Empty` rather than `default` if there are no null checks, and execute the code below unconditionally.